saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/TagModifyingJudgmentReferenceRemover.java

```diff
@@ -0,0 +1,128 @@
+package pl.edu.icm.saos.enrichment.reference;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import pl.edu.icm.saos.common.json.JsonStringParser;
+import pl.edu.icm.saos.common.json.JsonStringWriter;
+import pl.edu.icm.saos.persistence.enrichment.EnrichmentTagRepository;
+import pl.edu.icm.saos.persistence.enrichment.model.EnrichmentTag;
+import pl.edu.icm.saos.persistence.repository.JudgmentRepository;
+
+/**
+ * Class for removing reference to judgment from {@link EnrichmentTag}.
+ * It supports tags where reference removing is based on modifying {@link EnrichmentTag#getValue()}.
+ * 
+ * @author madryk
+ * @param <TAG_TYPE> - type of enrichment tag value type
+ */
+public abstract class TagModifyingJudgmentReferenceRemover<TAG_TYPE> implements JudgmentReferenceRemover {
+    
+    protected final static String JUDGMENT_IDS_QUERY_PARAM_NAME = "judgmentIds";
+    
+    @Autowired
+    private EntityManager entityManager;
+    
+    @Autowired
+    private EnrichmentTagRepository enrichmentTagRepository;
+    
+    @Autowired
+    private JudgmentRepository judgmentRepository;
+    
+    @Autowired
+    private JsonStringParser<TAG_TYPE> jsonStringParser;
+    
+    @Autowired
+    private JsonStringWriter<TAG_TYPE> jsonStringWriter;
+
+    
+    //------------------------ LOGIC --------------------------
+    
+    @Override
+    public void removeReference(List<Long> judgmentIds) {
+        
+        List<EnrichmentTag> results = findTagsWithReferences(judgmentIds);
+        
+        if (!results.isEmpty()) {
```

Comment:
In my opinion it is better to write 'if (results.isEmpty() {return;}' to get rid of nested curly braces.